File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
    if(nums.length == 0 || nums == null){ 
        return new int[]{};
    }

    int[] result = new int[nums.length]; 
    
    for(int i = 0; i < nums.length; i++){ 
    
    int self = nums[i]; 
    int prod = 1; 

    for(int j = 0; j < nums.length; j++){ 
        if(nums[j] != self && j != i){ 
        prod *= nums[j]; 
        }
    }

    result[i] = prod;
    }

    return result;
        
    }
}  

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        res = [] 

        counter = Counter(nums)
        lis = tuple(counter.items())

        lis = sorted(lis, key=lambda item: item[0])

        for i in range(k): 
            res.append(lis[i][0])

        return res


